lib/spack/spack/url.py

```diff
@@ -90,11 +90,16 @@ def find_list_url(url):
         (r'(.*github\.com/[^/]+/[^/]+)',
          lambda m: m.group(1) + '/releases'),
 
-        # GitLab
+        # GitLab non-API endpoint
         # e.g. https://gitlab.dkrz.de/k202009/libaec/uploads/631e85bcf877c2dcaca9b2e6d6526339/libaec-1.0.0.tar.gz
-        (r'(.*gitlab[^/]+/[^/]+/[^/]+)',
+        (r'(.*gitlab[^/]+/(?!api/v4/projects)[^/]+/[^/]+)',
          lambda m: m.group(1) + '/tags'),
 
+        # GitLab API endpoint
```

Comment:
Since we are only using GitLab API endpoints at the moment, can you put the API endpoint before the non-API endpoint in each function? Especially for the regexes, it will make things a bit faster.